File: dojo/mik.s/FizzBuzz/CSharp/05-02-2012 WhiteBelt/FizzBuzz/LinePrinter.cs
using System;
using System.Globalization;

namespace FizzBuzz {
    public class LinePrinter : ILinePrinter {
        public string PrintLine(uint lineNumber) {
            if (lineNumber == 0) {
                throw new ArgumentException();
            }

            var uses3 = (lineNumber % 3 == 0 || lineNumber.ToString(CultureInfo.InvariantCulture).Contains("3"));
            var uses5 = (lineNumber % 5 == 0 || lineNumber.ToString(CultureInfo.InvariantCulture).Contains("5"));

            if (uses3 && uses5) {
                return "fizzbuzz";
            } 
            
            if (uses5) {
                return "buzz";
            }
            
            if (uses3) {
                return "fizz";
            }

            return lineNumber.ToString(CultureInfo.InvariantCulture);
        }
    }
}

File: dojo/mik.s/FizzBuzz/Javascript/05-02-2012 WhiteBelt/src/LinePrinter.js
function LinePrinter() {
}

LinePrinter.prototype.printLine = function(lineNumber) {
	if (isNaN(lineNumber) || lineNumber <= 0) {
		throw new Error("lineNumber must be a number greater than zero");
	}
	
	var uses3 = lineNumber % 3 === 0 || String(lineNumber).indexOf('3') >= 0;
	var uses5 = lineNumber % 5 === 0 || String(lineNumber).indexOf('5') >= 0;
	
	if (uses3 && uses5) {
		return 'fizzbuzz';
	}
	
	if (uses5) {
		return 'buzz';
	}
	
	if (uses3) {
		return 'fizz';
	}
	
	return String(lineNumber);
}